crates/rslint_core/src/groups/errors/no_confusing_arrow.rs

```diff
@@ -0,0 +1,115 @@
+use crate::rule_prelude::*;
+use ast::{ArrowExpr, Expr};
+
+declare_lint! {
+    /**
+    Disallow arrow functions where they could be confused with comparisons.
+
+    Arrow functions (`=>`) are similar in syntax to some comparison operators (`>`, `<`, `<=`, and `>=`).
+    This rule warns against using the arrow function syntax in places where it could be confused with
+    a comparison operator
+
+    Here's an example where the usage of `=>` could be confusing:
+
+    ```js
+    // The intent is not clear
+    var x = a => 1 ? 2 : 3;
+    // Did the author mean this
+    var x = function (a) { return 1 ? 2 : 3 };
+    // Or this
+    var x = a >= 1 ? 2 : 3;
+    ```
+
+    ## Incorrect Code Examples
+
+    ```js
+    var x = a => 1 ? 2 : 3;
+    var x = (a) => 1 ? 2 : 3;
+    ```
+    */
+    #[serde(default)]
+    NoConfusingArrow,
+    errors,
+    "no-confusing-arrow",
+    /// Relaxes the rule and accepts parenthesis as a valid "confusion-preventing" syntax.
+    pub allow_parens: bool
+}
+
+impl Default for NoConfusingArrow {
+    fn default() -> Self {
+        Self { allow_parens: true }
+    }
+}
+
+#[typetag::serde]
+impl CstRule for NoConfusingArrow {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        let function_stmt = node.try_to::<ArrowExpr>()?;
+        let expr = Expr::cast(function_stmt.body()?.syntax().to_owned())?;
+
+        if is_conditional(&expr) && !(self.allow_parens && is_parenthesised(&expr)) {
+            let diagnostic = ctx
+                .err(
+                    self.name(),
+                    "Arrow function used ambiguously with a conditional expression",
```

Comment:
```suggestion
                    "arrow function in ternary expression could be mistaken for a comparison",
```
I believe this wording is more descriptive and does not use many terms that a user may not know